client/keys/add.go

```diff
@@ -226,20 +226,22 @@ func RunAddCmd(cmd *cobra.Command, args []string, kb keyring.Keyring, inBuf *buf
 	var mnemonic, bip39Passphrase string
 
 	recover, _ := cmd.Flags().GetBool(flagRecover)
-	if interactive || recover {
-		bip39Message := "Enter your bip39 mnemonic"
-		if !recover {
-			bip39Message = "Enter your bip39 mnemonic, or hit enter to generate one."
-		}
-
-		mnemonic, err = input.GetString(bip39Message, inBuf)
+	if recover {
+		mnemonic, err = input.GetString("Enter your bip39 mnemonic", inBuf)
 		if err != nil {
 			return err
 		}
-
 		if !bip39.IsMnemonicValid(mnemonic) {
 			return errors.New("invalid mnemonic")
 		}
+	} else if interactive {
+		mnemonic, err = input.GetString("Enter your bip39 mnemonic, or hit enter to generate one.", inBuf)
+		if err != nil {
+			return err
+		}
+		if !bip39.IsMnemonicValid(mnemonic) && mnemonic != "" {
```

Comment:
This should suffice:

```suggestion

				if !bip39.IsMnemonicValid(mnemonic) {
```

Reply:
it checks  for or "hit enter to generate one"
it has to be this way.

Reply:
have to exclude "" from invalid mnemonics, hope that explains the change?

Reply:
Yes @puneet2019, it makes sense. Thanks :+1: